Development/ios-objc-foundation-nmxcore Tests/Tests/C-Function Helpers/NMXLoggingTest.m

```diff
@@ -0,0 +1,217 @@
+//
+//  NMXLoggingTest.m
+//  ios-objc-foundation-nmxcore Tests
+//
+//  Created by Tobias Baube on 16.10.17.
+//  Copyright © 2017 Namics AG. All rights reserved.
+//
+
+#import <XCTest/XCTest.h>
+#import "NSString+NMXTest.h"
+#import "NSNumber+NMXTest.h"
+#import "NMXLogging.h"
+
+@interface NMXLoggingTest : XCTestCase
+@end
+
+@implementation NMXLoggingTest
+
+static int stderrSave;
+- (void)setUp {
+    [super setUp];
+    // Put setup code here. This method is called before the invocation of each test method in the class.
+    NSLog(@"Logs go to %@",[self logFilePath]);
+}
+
+- (void)tearDown {
+    // Put teardown code here. This method is called after the invocation of each test method in the class.
+    [super tearDown];
+}
+
+- (NSString *)logFilePath
+{
+    NSArray *allPaths = NSSearchPathForDirectoriesInDomains(NSDocumentDirectory, NSUserDomainMask, YES);
+    NSString *documentsDirectory = [allPaths objectAtIndex:0];
+    return [documentsDirectory stringByAppendingPathComponent:@"FWTestLog.txt"];
+}
+
+- (void)startLogging
+{
+    stderrSave = dup(STDERR_FILENO);
+    freopen([[self logFilePath] cStringUsingEncoding:NSASCIIStringEncoding],"a+",stderr);
+}
+
+- (void)closeLogging
+{
+    fflush(stderr);
+    dup2(stderrSave,STDERR_FILENO);
+    close(stderrSave);
+    NSFileManager *fileManager = [NSFileManager defaultManager];
+    NSError *error = nil;
+    [fileManager removeItemAtPath:[self logFilePath] error:&error];
+}
+
+
+- (BOOL)recordLogLogged:(NSString *)string
+{
+    NSString* content = [NSString stringWithContentsOfFile:[self logFilePath]
+                                                  encoding:NSUTF8StringEncoding
+                                                     error:NULL];
+    return [content hasSuffix:[string stringByAppendingString:@"\n"]];
+}
+
+- (void)testNMXLogFormat_Valid
+{
+//    XCTAssertNoThrow(NMXLog(nil));
```

Comment:
Did you forget to remove the commented lines?

Reply:
I forget to put these tests back in, thank you! Already pushed them.